src/templates/status.html

```diff
@@ -27,58 +49,205 @@
+    function getRunStatus(runId, cid, queryToken) {
+        function onSuccess({ runId, cid, jobId, queryToken, status }) {
+            for (var key in statusClassMap) {
+                if (status.indexOf(key) != -1) {
+                    if (key == SCHEDULED) {
+                        getQueuePosition(runId, jobId, cid, queryToken);
+                    } else {
+                        updateButtonContent(runId, status, "btn-" + statusClassMap[key]);
                     }
-                },
-                error: function () {
-                    updateButtonContent(runId, "Unknown. Click to try again.", "btn-warning");
+                    break;
                 }
-            })
+            }
+        }
+
+        function onError(param) {
+            updateButtonContent(runId, "Unknown. Click to try again.", "btn-warning");
+        }
+
+        getRunWithSingleJobHelper(runId, cid, queryToken, onSuccess, onError);
     }
 
-    function getQueuePosition(runId, jobId, cid, queryToken) {
+    /** Generate human readable display message given queue position as a string **/
+    function queuePositionTextGenerator(position) {
+        const posInt = parseInt(position);
+        if (isNaN(posInt)) {
+            return QUEUE_POSITION_QUERY_FAILED;
+        }
+        let positionString = `Your job is scheduled after ${posInt} other job${posInt > 1 ? "s" : ""}`;
+        if (posInt === 0) {
+            positionString = "Your job is next";
+        }
+        return positionString;
+    }
+
+    /** 
+     * Get queue position for a scheduled student job. Usage is similar to
+     * `getRunWithSingleJobHelper`.
+     */
+    function getQueuePositionHelper(runId, jobId, cid, queryToken, onSuccess, onError, extraParams = {}) {
         $.ajax(
             {
                 type: "get",
                 url: `{{ broadway_api_url }}/queue/${cid}/${jobId}/position`,
                 headers: buildHeader(queryToken),
                 success: function (result) {
-                    let position = result["data"]["position"]
-                    let queuePositionString = "Your job is scheduled after " + position + " other jobs";
-                    if (position == "0") {
-                        queuePositionString = "Your job is next";
-                    }
-                    updateButtonContent(runId, queuePositionString, "btn-info");
+                    let position = result["data"]["position"];
+                    onSuccess({runId, jobId, cid, queryToken, position, ...extraParams});
                 },
                 error: function () {
-                    updateButtonContent(runId, "Your run is scheduled. Queue position unknown.", "btn-info");
+                    onError({runId, jobId, cid, queryToken, ...extraParams});
                 }
             }
         )
     }
+    const QUEUE_POSITION_QUERY_FAILED = "Queue position query failed. Please refresh to try again";
```

Comment:
This may be overkill but you could put all strings like this into their own constant variables for consistency. Also, this variable is used on line `108` while it's defined down here. I think it makes more sense to define it at the top.